test/sys/test_wait.rs

```diff
@@ -6,6 +6,8 @@ use libc::exit;
 
 #[test]
 fn test_wait_signal() {
+    let m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
```

Comment:
You forgot to change this to `allow(unused_variables)` as well. I would suggest you also add a comment for each of these sections stating why we want to have an unused variable, as someone not familiar with this might think it should be removed without one.